app/assets/javascripts/application.js

```diff
@@ -375,7 +375,7 @@ $(function() {
 
 function update_puppetclasses(element) {
   var host_id = $("form").data('id')
-  var env_id = $('*[id*=environment_id]').val();
+  var env_id = $('*[name*=environment_id]').val();
```

Comment:
I have the feeling that since select2 we are not using the correct selector to extract the value... what do you think?